### PR TITLE
Add cloudwatch exporter

### PR DESCRIPTION
**Pull request by gianrubio**

### CODE REVIEW

File: incubator/cloudwatch-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+          volumeMounts:
+          - name: vol-cloudwatch-exporter
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/templates/deployment.yaml]: Please make list indentation consistent. This should indent one more level.

File: incubator/cloudwatch-exporter/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+description: A Helm chart for prometheus cloudwatch-exporter
+name: cloudwatch-exporter
+version: 0.1.0
+sources:
+- https://github.com/prometheus/cloudwatch_exporter
+keywords:
+- aws
+- cloudwatch
+- prometheus
+- exporter
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/Chart.yaml]: Add maintainer

File: incubator/cloudwatch-exporter/templates/serviceaccount.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.rbac.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "cloudwatch-exporter.fullname" . }}
+  labels:
+    app: {{ template "cloudwatch-exporter.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/templates/serviceaccount.yaml]: `chart: {{ template "cloudwatch-exporter.chart" . }}`

File: incubator/cloudwatch-exporter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# Default values for cloudwatch-exporter.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: prom/cloudwatch-exporter
+  tag: latest
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/values.yaml]: Can you specify a release version?

File: incubator/cloudwatch-exporter/values.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  pullPolicy: IfNotPresent
+
+service:
+  type: ClusterIP
+  port: 80
+  targetPort: 9100
+
+resources:
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/values.yaml]: Don't specify defaults for resources. Comment them out.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/cloudwatch-exporter/values.yaml (hunk 4 above)
Comment (gianrubio) [reply incubator/cloudwatch-exporter/values.yaml]: there's no other version for this project

**Comment by gianrubio** [comment]
@unguiculus all the requested changes are done

File: incubator/cloudwatch-exporter/values.yaml (hunk 4 above)
Comment (unguiculus) [reply incubator/cloudwatch-exporter/values.yaml]: Hm, that's bad.

File: incubator/cloudwatch-exporter/Chart.yaml [hunk 6]
```diff
@@ -10,3 +10,6 @@ keywords:
 - cloudwatch
 - prometheus
 - exporter
+maintainers:
+- email: gianrubio@gmail.com
+  name: Giancarlo Rubio
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/Chart.yaml]: Use Github username

File: incubator/cloudwatch-exporter/values.yaml [hunk 7]
```diff
@@ -14,13 +14,17 @@ service:
...
-    memory: 128Mi
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious 
+  # choice for the user. This also increases chances charts run on environments with little 
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following 
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #  cpu: 100m
+  #  memory: 128Mi
+  #requests:
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/values.yaml]: Space after #

File: incubator/cloudwatch-exporter/templates/clusterrole.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "cloudwatch-exporter.fullname" . }}
+  labels:
+    app: {{ template "cloudwatch-exporter.name" . }}
+    chart: {{ template "cloudwatch-exporter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/templates/clusterrole.yaml]: Indent one more level

File: incubator/cloudwatch-exporter/templates/clusterrolebinding.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "cloudwatch-exporter.fullname" . }}
+  labels:
+    app: {{ template "cloudwatch-exporter.name" . }}
+    chart: {{ template "cloudwatch-exporter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+subjects:
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/templates/clusterrolebinding.yaml]: Indent one more level

File: incubator/cloudwatch-exporter/README.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+| `imageTag`                      | Image tag                                  | `latest`                                      |
+| `imagePullPolicy`               | Image pull policy                          | `IfNotPresent` |
+| `service.type`      | Service type |  `ClusterIP` | 
+| `service.port`                      | The service port                               | `80`                                     |
+| `service.targetPort`                      | The target port of the container                               | `9100`                                        |
+| `resources.limits.cpu`          | CPU limit                                  |                                                     |
+| `resources.limits.memory`       | Memory limit                               |                                                     |
+| `resources.requests.cpu`        | CPU request                                |                                                    |
+| `resources.requests.memory`     | Memory request                             |                                                  |
```
Comment (unguiculus) [inline incubator/cloudwatch-exporter/README.md]: I'd suggest you only add one line for resources, the default being `{}`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by gianrubio** [comment]
@unguiculus I'm going to add other prometheus exporters like Postgres, elasticsearch... Should I rename this to prometheus-cloudwatch-exporter

**Comment by unguiculus** [comment]
Renaming would be a good idea. You may add them to the stable repo directly and also move this one.

**Comment by gianrubio** [comment]
@unguiculus done, please review it again

**Comment by gianrubio** [comment]
ping @unguiculus

File: stable/prometheus-cloudwatch-exporter/README.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the Cloudwatch Exporter chart and their default values.
+
+| Parameter                       | Description                                | Default                                                    |
```
Comment (unguiculus) [inline stable/prometheus-cloudwatch-exporter/README.md]: Please format the table properly. Either align columns or completely remove excess whitespace.

File: stable/prometheus-cloudwatch-exporter/templates/serviceaccount.yaml [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.rbac.create }}
```
Comment (unguiculus) [inline stable/prometheus-cloudwatch-exporter/templates/serviceaccount.yaml]: `{{- if .Values.serviceAccount.create }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**